src/export.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::{Result, Context, anyhow};
+use clap::Args;
+use std::path::Path;
+use crate::{
+    account::derive_and_cache_addresses,
+    utils::{display_string_discreetly, load_wallet},
+    DEFAULT_CACHE_ACCOUNTS,
+};
+
+#[derive(Debug, Args)]
+pub struct Export {
+    /// Forces export even if it might be unsafe
+    #[clap(short, long)]
+    pub force: bool,
+    /// How many accounts to cache by default (Default 10)
+    #[clap(short, long)]
+    pub cache_accounts: Option<usize>,
+}
+
+/// Decrypts a wallet using provided password
+fn decrypt_wallet(wallet_path: &Path, password: &str) -> Result<String> {
+    let phrase_bytes = eth_keystore::decrypt_key(wallet_path, password)
+        .map_err(|e| anyhow!("Failed to decrypt keystore: {}", e))?;
+    
+    String::from_utf8(phrase_bytes)
+        .context("Invalid UTF-8 in mnemonic phrase")
+}
+
+/// Prompts for password securely
+fn prompt_password() -> Result<String> {
+    const PROMPT: &str = "Please enter your wallet password to export the mnemonic phrase: ";
+    rpassword::prompt_password(PROMPT)
+        .map_err(|e| anyhow!("Password prompt error: {}", e))
+}
+
+/// Displays mnemonic in alternate screen
+fn display_mnemonic(phrase: &str) -> Result<()> {
+    let mnemonic_string = format!("Mnemonic phrase: {}\n", phrase);
+    display_string_discreetly(&mnemonic_string, "### Press any key to complete. ###")
+}
+
+/// Securely wipes sensitive data from memory
+fn secure_wipe(data: &mut [u8]) {
```

Comment:
Good idea! We should use [zeroize](https://crates.io/crates/zeroize) instead to make sure this isn't optimized away by the compiler.